rsconnect/bundle.py

```diff
@@ -1245,6 +1433,89 @@ def write_notebook_manifest_json(
+    :param app_mode: the application mode to assume.  If this is None, the extension
+    portion of the entry point file name will be used to derive one. Previous default = None.
+    :param extra_files: any extra files that should be included in the manifest. Previous default = None.
+    :param excludes: a sequence of glob patterns that will exclude matched files.
+    :param force_generate: bool indicating whether to force generate manifest and related environment files.
+    :param image: the optional docker image to be specified for off-host execution. Default = None.
+    :return: the manifest data structure.
+    """
+    extra_files = list(extra_files) if extra_files else []
+    excludes = list(excludes) if excludes else []
+
+    entrypoint = entrypoint or infer_entrypoint(path=path, mimetype="text/ipynb")
+    base_dir = dirname(entrypoint)
+    manifest = Manifest(app_mode=AppModes.JUPYTER_VOILA, environment=environment, entrypoint=entrypoint, image=image)
+
+    if isfile(path):
+        validate_file_is_notebook(entrypoint)
+        manifest.entrypoint = entrypoint
+    else:
+        manifest.entrypoint = "" if not entrypoint else entrypoint  # multi-notebook mode
+
+    # handle environment files
+    if not exists(join(base_dir, environment.filename)) or force_generate:
+        manifest.add_file(join(base_dir, environment.filename))
+
+    excludes.extend(["manifest.json"])
+    file_list = create_file_list(path, extra_files, excludes)
+    for rel_path in file_list:
+        manifest.add_file(rel_path)
+    return manifest
+
+
+def write_voila_manifest_json(
+    path: str,
+    entrypoint: str,
+    environment: Environment,
+    app_mode: AppMode = AppModes.JUPYTER_VOILA,
+    extra_files: typing.List[str] = None,
+    excludes: typing.List[str] = None,
+    force_generate: bool = True,
+    image: str = None,
+) -> bool:
+    """
+    Creates and writes a manifest.json file for the given path.
+
+    :param path: the file, or the directory containing the files to deploy.
+    :param entry_point: the main entry point for the API.
+    :param environment: the Python environment to start with.  This should be what's
+    returned by the inspect_environment() function.
+    :param app_mode: the application mode to assume.  If this is None, the extension
+    portion of the entry point file name will be used to derive one. Previous default = None.
+    :param extra_files: any extra files that should be included in the manifest. Previous default = None.
+    :param excludes: a sequence of glob patterns that will exclude matched files.
+    :param force_generate: bool indicating whether to force generate manifest and related environment files.
+    :param image: the optional docker image to be specified for off-host execution. Default = None.
+    :return: whether the manifest was written.
+    """
+    manifest = create_voila_manifest(**locals())
+    entrypoint = entrypoint or infer_entrypoint(path=path, mimetype="text/ipynb")
```

Comment:
Could the manifest entrypoint be used here?

Reply:
The manifest can change (make all files and metadata relative to the entrypoint) so using the original path is safer here.